packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderContent/ConditionBuilderContent.js

```diff
@@ -0,0 +1,83 @@
+import React, { useContext, useEffect, useRef, useState } from 'react';
+import { pkg } from '../../../settings';
+import { Button } from '@carbon/react';
+import { Add } from '@carbon/react/icons';
+import {
+  ConditionBuilderContext,
+  emptyState,
+} from '../ConditionBuilderContext/DataTreeContext';
+
+const blockClass = `${pkg.prefix}--condition-builder`;
+
+function ConditionBuilderContent() {
+  const { rootState, setRootState } = useContext(ConditionBuilderContext);
+  const [isConditionbuilderActive, setIsConditionbuilderActive] =
+    useState(true);
+  const conditionBuilderRef = useRef();
+
+  useEffect(() => {
+    if (!rootState?.groups || rootState?.groups.length == 0) {
+      setIsConditionbuilderActive(true);
+    } else {
+      setIsConditionbuilderActive(false);
+      if (
+        rootState.groups[0].conditions.length == 1 &&
+        rootState.groups[0].conditions[0].property == undefined
+      ) {
+        // when the add condition clicked to start the condition building, we by default open the popover of the first property
+        setTimeout(() => {
+          conditionBuilderRef.current.querySelector('.propertyField').click();
+        }, 0);
+      }
+    }
+  }, [rootState]);
+
+  const onStartConditionbuilder = () => {
+    //when add conditon button is clicked.
+    setIsConditionbuilderActive(false);
+    setRootState(emptyState); //here we can set an empty skeleton object for an empty condition builder,
+    //or we can even prepopulate some existing builder and continue editing
+  };
+
+  return (
+    <div ref={conditionBuilderRef}>
+      {isConditionbuilderActive && (
+        <Button
+          className={`${blockClass}__condition-builder`}
+          renderIcon={(props) => <Add size={16} {...props} />}
+          iconDescription={'Add Condition'}
+          kind="ghost"
+          size="sm"
+          onClick={() => onStartConditionbuilder()}
```

Comment:
no need to pass it is as an anonymous function `onClick={onStartConditionbuilder}`

Reply:
modified as advised